### ISSUE TITLE
 Can not perform this action after onSaveInstanceState

### ISSUE THREAD

**Issue by thelong1EU** [firebase/FirebaseUI-Android#857]
* Android device: Nexus 9
  * Android OS version: 25
  * Firebase/Play Services SDK version: 11.0.4
  * FirebaseUI version: 2.2.0

#### Steps to reproduce:

  1. go to home screen when the verification dialog is on screen

  * What happened?  This could be a description, `logcat` output, etc.
 ```
 Fatal Exception: java.lang.IllegalStateException: Can not perform this action after onSaveInstanceState
       at android.support.v4.app.FragmentManagerImpl.checkStateLoss(FragmentManager.java:1884)
       at android.support.v4.app.FragmentManagerImpl.enqueueAction(FragmentManager.java:1902)
       at android.support.v4.app.BackStackRecord.commitInternal(BackStackRecord.java:650)
       at android.support.v4.app.BackStackRecord.commit(BackStackRecord.java:609)
       at android.support.v4.app.DialogFragment.dismissInternal(DialogFragment.java:205)
       at android.support.v4.app.DialogFragment.dismiss(DialogFragment.java:171)
       at com.firebase.ui.auth.ui.phone.PhoneVerificationActivity.dismissLoadingDialog(PhoneVerificationActivity.java:390)
       at com.firebase.ui.auth.ui.phone.PhoneVerificationActivity.access$000(PhoneVerificationActivity.java:56)
       at com.firebase.ui.auth.ui.phone.PhoneVerificationActivity$1.run(PhoneVerificationActivity.java:189)
       at android.os.Handler.handleCallback(Handler.java:751)
       at android.os.Handler.dispatchMessage(Handler.java:95)
       at android.os.Looper.loop(Looper.java:154)
       at android.app.ActivityThread.main(ActivityThread.java:6119)
       at java.lang.reflect.Method.invoke(Method.java)
       at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:886)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:776)
```

**Comment by samtstern** [comment]
@SUPERCILEX this is probably related to #817, where the dialog was changed to a `DialogFragment`.  I've seen this before, it has to do with the fact that dialog fragments don't have `commitAllowingStateLoss()` and also the fact that Fragments are truly a creation of demons meant to torment Android developers for all time.

**Comment by thelong1EU** [comment]
:rofl:

**Comment by SUPERCILEX** [comment]
@samtstern Haha 😆 Ok, I'll look into it.

**Comment by samtstern** [comment]
This has been fixed and released in `2.3.0`.
